src/node.rs

```diff
@@ -101,13 +103,21 @@ impl<S: StorageExt> SawtoothRaftNode<S> {
             self.leader_state = Some(LeaderState::Validating(block.block_id.clone()));
         }
 
+        debug!("Block has been received: {:?}", &block);
+        self.new_block_backlog.push_back(block.clone());
+
         self.service.check_blocks(vec![block.block_id]).expect("Failed to send check blocks");
     }
 
     pub fn on_block_valid(&mut self, block_id: BlockId) {
         // Handle out of order new/valid updates
         debug!("Block has been validated: {:?}", &block_id);
-        self.valid_block_backlog.push_back(block_id.clone());
+        let block = self.new_block_backlog
+            .iter()
+            .cloned()
+            .find(|block| block.block_id == block_id)
+            .expect("BlockNew not received before BlockValid");
+        self.valid_block_backlog.push_back(block);
```

Comment:
What is the consequence of failure here? Will the event be regenerated later?

Reply:
It could be a protocol error. I would suggest logging as error and returning. Otherwise, the whole consensus engine halts

Reply:
We can fix this in a subsequent PR, though.

Reply:
If it's brief would prefer we just fix this PR.